extensions/mssql/src/dashboard/bookWidget.ts

```diff
@@ -73,56 +69,6 @@ export function registerBooksWidget(bookContributionProvider: BookContributionPr
 	});
 }
 
-async function promptForFolder(bookContribution: BookContribution): Promise<void> {
-	try {
-		const allFilesFilter = localize('allFiles', "All Files");
-		let filter = {};
-		filter[allFilesFilter] = '*';
-		let uris = await vscode.window.showOpenDialog({
-			filters: filter,
-			canSelectFiles: false,
-			canSelectMany: false,
-			canSelectFolders: true,
-			openLabel: localize('labelPickFolder', "Pick Folder")
-		});
-		if (uris && uris.length > 0) {
-			let pickedFolder = uris[0];
-			let destinationUri: vscode.Uri = vscode.Uri.file(path.join(pickedFolder.fsPath, bookContribution.name));
-			await saveBooksToFolder(destinationUri, bookContribution);
-			await promptToReloadWindow(destinationUri);
-		}
-		return;
-	} catch (error) {
-		vscode.window.showErrorMessage(localize('FailedDuringSaveAndPrompt', 'Failed : {0}', Utils.getErrorMessage(error)));
-	}
-}
-
-async function saveBooksToFolder(folderUri: vscode.Uri, bookContribution: BookContribution): Promise<void> {
-	// Get book contributions
-	if (bookContribution && folderUri) {
-		//remove folder if exists
-		await fs.removeSync(folderUri.fsPath);
-		//make directory for each contribution book.
-		await fs.mkdirSync(folderUri.fsPath);
-		await fs.copy(bookContribution.path, folderUri.fsPath);
-	}
+function openBookViewlet(folderUri: vscode.Uri): void {
```

Comment:
I think i’d rather have the contribution decide whether or not it should open as readonly or not. Can we make that happen?

Reply:
that can be done. Created an issue to track this:https://github.com/microsoft/azuredatastudio/issues/6841